site/posts/writing-multiple-netcdf-files-in-parallel-with-xarray-and-dask.ipynb

```diff
@@ -0,0 +1,2084 @@
+       "     yc       (y, x) float64 dask.array<chunksize=(205, 275), meta=np.ndarray>\n",
+       " Dimensions without coordinates: x, y\n",
+       " Data variables:\n",
+       "     Tair     (time, y, x) float64 dask.array<chunksize=(12, 205, 275), meta=np.ndarray>\n",
+       " Attributes:\n",
+       "     title:                     /workspace/jhamman/processed/R1002RBRxaaa01a/l...\n",
+       "     institution:               U.W.\n",
+       "     source:                    RACM R1002RBRxaaa01a\n",
+       "     output_frequency:          daily\n",
+       "     output_mode:               averaged\n",
+       "     convention:                CF-1.4\n",
+       "     references:                Based on the initial model of Liang et al., 19...\n",
+       "     comment:                   Output from the Variable Infiltration Capacity...\n",
+       "     nco_openmp_thread_number:  1\n",
+       "     NCO:                       \"4.6.0\"\n",
+       "     history:                   Tue Dec 27 14:15:22 2016: ncatted -a dimension...)"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "x[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Each item in the returned list consists of a tuple of a dictionary and an xarray dataset corresponding to sliced chunk. The dictionary contains information about how the original xarray dataset was sliced. Let's extract the xarray datasets into a standalone list:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datasets = [item[1] for item in x]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we have a list of datasets, we need another helper function that will generate filenames for us when given an xarray dataset as input:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create a helper function for generating filepath"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_filepath(ds, prefix='filename', root_path=\".\"):\n",
+    "    \"\"\"\n",
+    "    Generate a filepath when given an xarray dataset\n",
+    "    \"\"\"\n",
+    "    start = ds.time.data[0].strftime(\"%Y-%m-%d\")\n",
+    "    end = ds.time.data[-1].strftime(\"%Y-%m-%d\")\n",
+    "    filepath = f'{root_path}/{prefix}_{start}_{end}.nc'\n",
+    "    return filepath"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'./filename_1981-09-16_1982-08-17.nc'"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# test that our function is working\n",
```

Comment:
Make this code comment a text cell above this code cell (and make it a complete sentence).

Reply:
Done!